Experiment automation/(LUMICKS) Reference kit automation/referencekit_automation.py

```diff
@@ -0,0 +1,335 @@
+    attempts = 0
+    while f_current < force_threshold:
+        if attempts > fishing_attempts:
+            raise RuntimeError(f"Max DNA fishing attempts {fishing_attempts} reached.")
+
+        print(f"Fishing for DNA: attempt {attempts}/{fishing_attempts}")
+        goto_distance(min_distance, match_threshold, speed=fishing_speed)
+        bl.pause(0.5)
+        goto_distance(max_distance, match_threshold, speed=fishing_speed)
+        f_current = get_force(dt=dt)
+
+        attempts += 1
+
+
+def make_kymograph(force_kymo, match_threshold, name, path):
+    goto_force(force_kymo, match_threshold)
+    try:
+        goto_force(force_kymo, match_threshold)
+        t0 = bl.timeline.now
+        bl.confocal.start_scan()
+        print("start kymograph")
+        bl.pause(10.0)
+        print(f"go to a force of {force_kymo} pN")
+        goto_force(force_kymo, match_threshold, speed=1)
+        bl.pause(kymo_duration)
+    finally:
+        print("Stop kymograph")
+        bl.confocal.abort_scan()
+        bl.pause(2)
+        t1 = bl.timeline.now
+        marker = bl.Marker(name, start = t0, stop = t1)
+        marker.export(Path(path))
+    bl.pause(1)
+
+
+def validate_dir(path):
+    """Make sure the user entered a valid path"""
+    path = path if path[-1] == '/' else path + '/'
+    if os.path.isdir(path):
+        return path
+    else:
+        raise RuntimeError("Invalid path specified. Did you pass a directory name?")
+
+
+def check_tether_breakage(test_force):
+    if get_force() < test_force:
+        raise RuntimeError("Lost tether")
+
+
+def check_multiple_tethers(max_distance, multiple_tether_force, force_threshold, match_threshold, max_retries, speed, dna_length):
+    """Check whether there are multiple tethers by going to a large distance and checking the force.
+
+    Parameters:
+    max_distance: The maximum distance to go to check for multiple tethers [um]
+    multiple_tether_force: The force threshold above which we assume there are multiple tethers [pN]
+    force_threshold: The force threshold above which we assume there is still a tether [pN]
+    match_threshold: The threshold for the match score to check whether the beads are still caught
+    max_retries: The maximum number of attempts to break tethers
+    speed: The speed at which to move the trap [um/s]
+    """
+    goto_distance(0.5 * dna_length, match_threshold, speed=speed)
+    bl.reset_force()
+    goto_distance(0.95 * dna_length, match_threshold, speed=speed)
+    if get_force() > multiple_tether_force:
+        print("Caught multiple DNA tethers, let's try to break tethers until there is only one left")
+        step_size = 0.1 * dna_length
+        one_tether_remains = False
+        n = 0
+        while not one_tether_remains:
+            if n == max_retries:
+                raise RuntimeError(f"Required more than {max_retries} attempts to break tethers")
+            """Go to a large distance, trying to break tethers:"""

```

Comment:
Bit unusual to use multi-line comments for single line comments.
```suggestion
            # Go to a large distance, trying to break tethers:
```